logfire/integrations/pydantic.py

```diff
@@ -292,61 +291,69 @@ def get_schema_name(schema: CoreSchema) -> str:
 class LogfirePydanticPlugin:
     """Implements a new API for pydantic plugins.
 
-    Patches pydantic to accept this new API shape.
-
-    Set the `LOGFIRE_DISABLE_PYDANTIC_PLUGIN` environment variable to `true` to disable the plugin.
-
-    Note:
-        In the future, you'll be able to use the `PYDANTIC_DISABLE_PLUGINS` instead.
+    Patches Pydantic to accept this new API shape.
 
-        See [pydantic/pydantic#7709](https://github.com/pydantic/pydantic/issues/7709) for more information.
+    Set the `LOGFIRE_PYDANTIC_RECORD` environment variable to `"off"` to disable the plugin, or
+    `PYDANTIC_DISABLE_PLUGINS` to `true` to disable all Pydantic plugins.
     """
 
-    def new_schema_validator(
-        self,
-        schema: CoreSchema,
-        schema_type: Any,
-        schema_type_path: SchemaTypePath,
-        schema_kind: SchemaKind,
-        config: CoreConfig | None,
-        plugin_settings: dict[str, Any],
-    ) -> tuple[_ValidateWrapper, ...] | tuple[None, ...]:
-        """This method is called every time a new `SchemaValidator` is created.
-
-        Args:
-            schema: The schema to validate against.
-            schema_type: The original type which the schema was created from, e.g. the model class.
-            schema_type_path: Path defining where `schema_type` was defined, or where `TypeAdapter` was called.
-            schema_kind: The kind of schema to validate against.
-            config: The config to use for validation.
-            plugin_settings: The plugin settings.
-
-        Returns:
-            A tuple of decorator factories for each of the three validation methods -
-                `validate_python`, `validate_json`, `validate_strings` or a tuple of
-                three `None` if recording is `off`.
-        """
-        # Patch a bug that occurs even if the plugin is disabled.
-        _patch_PluggableSchemaValidator()
-
-        logfire_settings = plugin_settings.get('logfire')
-        if logfire_settings and 'record' in logfire_settings:
-            record = logfire_settings['record']
-        else:
-            record = _pydantic_plugin_config().record
+    if get_version(pydantic.__version__) < get_version('2.5.0'):  # pragma: no cover
+
+        def new_schema_validator(  # type: ignore[reportRedeclaration]
+            self, schema: CoreSchema, config: CoreConfig | None, plugin_settings: dict[str, Any]
+        ) -> tuple[_ValidateWrapper, ...] | tuple[None, ...]:
```

Comment:
```suggestion
    if get_version(pydantic.__version__) < get_version('2.5.0') or os.environ.get('LOGFIRE_PYDANTIC_RECORD') == 'off':  # pragma: no cover

        def new_schema_validator(  # type: ignore[reportRedeclaration]
            self, *_: Any, *__: Any
        ) -> tuple[_ValidateWrapper, ...] | tuple[None, ...]:
```

Best to make `LOGFIRE_PYDANTIC_RECORD=off` a complete no-op in case of bugs such as https://pydanticlogfire.slack.com/archives/C06EDRBSAH3/p1714528611959999.